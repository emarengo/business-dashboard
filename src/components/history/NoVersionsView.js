import React from 'react';

import styled from '@emotion/styled';

import { ReactComponent as NoVersionsSvg } from '../../assets/no-versions.svg';

const NoVersionsLogo = styled(NoVersionsSvg)`
  margin-bottom: 30px;
`;
const NoVersionsContainer = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(to bottom, #dde6f2 0%, #f2f6fa);
  color: #093548;
  font-family: 'Jost', sans-serif;
  z-index: 5;
`;

const NoVersionsTitle = styled.h1`
  font-size: 25px;
  font-weight: 800;

  margin-bottom: 36px;
`;

const NoVersionsText = styled.p`
  width: 412px;
  font-size: 17px;
  line-height: 1.31;
  font-weight: 400;
  text-align: center;
`;

const NoVersionsView = () => {
  return (
    <NoVersionsContainer>
      <NoVersionsLogo data-testid="noVersionsLogo" />
      <NoVersionsTitle data-testid="noVersionsTitle">
        No versions found
      </NoVersionsTitle>
      <NoVersionsText data-testid="noVersionsText">
        It seems that this city hasn’t any older versions or geofenced area
        hasn’t been created yet. Try another city.
      </NoVersionsText>
    </NoVersionsContainer>
  );
};

export default NoVersionsView;
