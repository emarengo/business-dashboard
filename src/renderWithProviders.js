import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'emotion-theming';
import { BeatTheme } from '@thebeatapp/beat-ui';

const customRender = (node, ...options) => {
  return render(
    <Router>
      <ThemeProvider theme={BeatTheme}>{node}</ThemeProvider>
    </Router>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
export { customRender as rerender };
