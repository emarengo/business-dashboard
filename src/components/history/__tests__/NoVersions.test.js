import React from 'react';
import { cleanup, render } from '../../../renderWithProviders';
import NoVersionsView from '../NoVersionsView';

describe('No Versions Screen', () => {
  let renderResult;
  const titleText = 'No versions found';
  const titleDescription =
    'It seems that this city hasn’t any older versions or geofenced area hasn’t been created yet. Try another city';

  beforeEach(() => {
    renderResult = render(<NoVersionsView />);
  });

  afterEach(cleanup);

  it('should display logo image', () => {
    const logo = renderResult.getByTestId('noVersionsLogo');
    expect(logo).toBeInTheDocument();
  });

  it('should show correct title', () => {
    const title = renderResult.getByTestId('noVersionsTitle');
    expect(title).toHaveTextContent(titleText);
  });

  it('should show correct description', () => {
    const description = renderResult.getByTestId('noVersionsText');
    expect(description).toHaveTextContent(titleDescription);
  });

  it('should match screenshot', () => {
    expect(renderResult).toMatchSnapshot();
  });
});
