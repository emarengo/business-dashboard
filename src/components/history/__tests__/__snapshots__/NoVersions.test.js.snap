// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Versions Screen should match screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      class="beat-ui-portal"
    />
    <div>
      <div
        class="css-ll51bc"
      >
        <svg
          class="css-1iehrax"
          data-testid="noVersionsLogo"
        >
          no-versions.svg
        </svg>
        <h1
          class="css-1uf66b1"
          data-testid="noVersionsTitle"
        >
          No versions found
        </h1>
        <p
          class="css-1rfud7a"
          data-testid="noVersionsText"
        >
          It seems that this city hasn’t any older versions or geofenced area hasn’t been created yet. Try another city.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-ll51bc"
    >
      <svg
        class="css-1iehrax"
        data-testid="noVersionsLogo"
      >
        no-versions.svg
      </svg>
      <h1
        class="css-1uf66b1"
        data-testid="noVersionsTitle"
      >
        No versions found
      </h1>
      <p
        class="css-1rfud7a"
        data-testid="noVersionsText"
      >
        It seems that this city hasn’t any older versions or geofenced area hasn’t been created yet. Try another city.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
